[2017-11-21 01:49:21,921] olympus.Olympus<Olympus:e4002>:OUTPUT: Parsed config: {'clients': {}, 'replicas': {}, 'failures': {0: {'r2': {'shuttle': [{'failure': ('invalid_result_sig', ''), 'trigger_point': {'event_num': 2, 'application': 'post', 'client_id': 'c0'}}]}}}, 'test_case_name': 'test_invalid_result_sig_t1_single_client', 't': 1, 'num_client': 1, 'client_timeout': 3000, 'head_timeout': 3000, 'nonhead_timeout': 3000, 'checkpt_interval': 100, 'hosts': {0: 'localhost;'}, 'client_hosts': {'c0': 'localhost;'}, 'replica_hosts': {'r0': '0', 'r1': '0', 'r2': '0'}, 'workloads': {'c0': ["put('movie','star')", "get('movie')", "append('movie', '123')", "get('movie')", "get('movie')", "get('movie')", "get('movie')", "get('movie')", "get('movie')", "get('movie')", "get('movie')", "get('movie')", "get('movie')", "get('movie')", "get('movie');"]}}
[2017-11-21 01:49:21,926] olympus.Olympus<Olympus:e4002>:OUTPUT: Generated keys: {'private_keys': {'o1': b'f95b198677142db5d0b697092989d9eef350ed98f9a5b400074f306bf86e5321', 'c0': b'2fe4f6fe9638e39f6c2d809719f022fe08c1ac0840ee0f543b213da75c716943', 'r0': b'01aa9be1483a5af6e41cad5752f25584de5d5152143f28d18f4873bf3d9c90ba', 'r1': b'4292ba0bceb92bfc01937d664a5f75a9a6c1231af80dea358053170d942a6531', 'r2': b'9c5b8afec9c9506850a5a1c17c6a52615ca45d153183cdcb1de28235435f8eb1'}, 'public_keys': {'o1': b'89282bef18ca1d602fc431b71e1a837e7b6b64370ffe0d42681b25f34a3e9268', 'c0': b'983b0b24cb98f56d0e209e49b13cbf77c39c0b0caff1d2613eaf767b63643c1f', 'r0': b'580687a4cc150f7fca428cb2ff54267d70a9e0bf5ac5134b3610234d914d08c9', 'r1': b'36325269ed09e8141e14dc0875518f2e5cd31506810aec6896c679ea64db0a20', 'r2': b'239990130b1c935a4be27f9c3885d93151fe2f4111a56502914bd37d437c7eab'}}
[2017-11-21 01:49:22,163] olympus.Olympus<Olympus:e4002>:OUTPUT: Initializing cluster with configuration version:  0
[2017-11-21 01:49:22,920] replica.Replica<Replica:98002>:OUTPUT: Spawned and initialized replica:  r0
[2017-11-21 01:49:22,925] replica.Replica<Replica:98003>:OUTPUT: Spawned and initialized replica:  r1
[2017-11-21 01:49:22,930] replica.Replica<Replica:98004>:OUTPUT: Spawned and initialized replica:  r2
[2017-11-21 01:49:22,935] client.Client<Client:98001>:OUTPUT: c0  initialized with config= {'id': 'c0', 'workload': ["put('movie','star')", "get('movie')", "append('movie', '123')", "get('movie')", "get('movie')", "get('movie')", "get('movie')", "get('movie')", "get('movie')", "get('movie')", "get('movie')", "get('movie')", "get('movie')", "get('movie')", "get('movie');"], 'private_key': b'2fe4f6fe9638e39f6c2d809719f022fe08c1ac0840ee0f543b213da75c716943', 'public_keys': {'o1': b'89282bef18ca1d602fc431b71e1a837e7b6b64370ffe0d42681b25f34a3e9268', 'c0': b'983b0b24cb98f56d0e209e49b13cbf77c39c0b0caff1d2613eaf767b63643c1f', 'r0': b'580687a4cc150f7fca428cb2ff54267d70a9e0bf5ac5134b3610234d914d08c9', 'r1': b'36325269ed09e8141e14dc0875518f2e5cd31506810aec6896c679ea64db0a20', 'r2': b'239990130b1c935a4be27f9c3885d93151fe2f4111a56502914bd37d437c7eab'}, 'client_timeout': 3.0, 'olympus': {'id': 'o1', 'process_id': <Olympus:e4002>}, 'replicas': {'r0': <Replica:98002>, 'r1': <Replica:98003>, 'r2': <Replica:98004>}, 'head_id': 'r0', 't': 1}
[2017-11-21 01:49:22,939] client.Client<Client:98001>:OUTPUT: The following operation was created: {'id': 0, 'client_id': 'c0', 'command_string': "put('movie','star')"}
[2017-11-21 01:49:22,939] client.Client<Client:98001>:OUTPUT: return_value= OK
[2017-11-21 01:49:22,943] client.Client<Client:98001>:OUTPUT: A request for an operation was made {'content': {'is_retry': False, 'operation': {'id': 0, 'client_id': 'c0', 'command_string': "put('movie','star')"}}, 'signed_content': b"M\x9cs\xa7\xff&\n\xc0f\x9ed\x89\xea\x089\xc2\xfc\xcb\xb0w0\xcej\xbd\x10\xbeI\x17\xdc\x99\xe1\x90\x01*=5\xf1h\xdfMv\x8d\xce7\x079\xa9F\xdd\x01-\xb6\xf2\xab\xd58h\xff\xd7b\xb8j0\x03\x80\x03}q\x00(X\x08\x00\x00\x00is_retryq\x01\x89X\t\x00\x00\x00operationq\x02}q\x03(X\x02\x00\x00\x00idq\x04K\x00X\t\x00\x00\x00client_idq\x05X\x02\x00\x00\x00c0q\x06X\x0e\x00\x00\x00command_stringq\x07X\x13\x00\x00\x00put('movie','star')q\x08uu."}
[2017-11-21 01:49:22,944] replica.Replica<Replica:98002>:OUTPUT: Received request message (source, client_id, operation_id, self_id)  c0 c0 0 r0
[2017-11-21 01:49:22,944] replica.Replica<Replica:98002>:OUTPUT: Checking for failure associated with trigger, application:  ('client_request', 'pre')  in replica:  r0
[2017-11-21 01:49:22,945] replica.Replica<Replica:98002>:OUTPUT: No failure associated with trigger:  client_request  found in replica:  r0
[2017-11-21 01:49:22,948] replica.Replica<Replica:98002>:OUTPUT: Head is processing request (operation_id, client_id) (0, 'c0')
[2017-11-21 01:49:22,952] replica.Replica<Replica:98002>:OUTPUT: Checking for failure associated with trigger, application:  ('client_request', 'post')  in replica:  r0
[2017-11-21 01:49:22,953] replica.Replica<Replica:98002>:OUTPUT: No failure associated with trigger:  client_request  found in replica:  r0
[2017-11-21 01:49:22,959] replica.Replica<Replica:98003>:OUTPUT: Received shuttle (source, client_id, operation_id, self_id)  r0 c0 0 r1
[2017-11-21 01:49:22,959] replica.Replica<Replica:98003>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'pre')  in replica:  r1
[2017-11-21 01:49:22,960] replica.Replica<Replica:98003>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r1
[2017-11-21 01:49:22,972] replica.Replica<Replica:98003>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'post')  in replica:  r1
[2017-11-21 01:49:22,973] replica.Replica<Replica:98003>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r1
[2017-11-21 01:49:22,980] replica.Replica<Replica:98004>:OUTPUT: Received shuttle (source, client_id, operation_id, self_id)  r1 c0 0 r2
[2017-11-21 01:49:22,980] replica.Replica<Replica:98004>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'pre')  in replica:  r2
[2017-11-21 01:49:22,981] replica.Replica<Replica:98004>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r2
[2017-11-21 01:49:23,009] replica.Replica<Replica:98004>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'post')  in replica:  r2
[2017-11-21 01:49:23,010] replica.Replica<Replica:98004>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r2
[2017-11-21 01:49:23,014] replica.Replica<Replica:98003>:OUTPUT: Received result shuttle (source, client_id, operation_id, self_id)  r2 c0 0 r1
[2017-11-21 01:49:23,015] replica.Replica<Replica:98004>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'post')  in replica:  r2
[2017-11-21 01:49:23,015] replica.Replica<Replica:98003>:OUTPUT: Checking for failure associated with trigger, application:  ('result_shuttle', 'pre')  in replica:  r1
[2017-11-21 01:49:23,015] replica.Replica<Replica:98004>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r2
[2017-11-21 01:49:23,015] replica.Replica<Replica:98003>:OUTPUT: No failure associated with trigger:  result_shuttle  found in replica:  r1
[2017-11-21 01:49:23,020] client.Client<Client:98001>:OUTPUT: A result was received from replica= r2
[2017-11-21 01:49:23,031] replica.Replica<Replica:98003>:OUTPUT: Checking for failure associated with trigger, application:  ('result_shuttle', 'post')  in replica:  r1
[2017-11-21 01:49:23,032] replica.Replica<Replica:98003>:OUTPUT: No failure associated with trigger:  result_shuttle  found in replica:  r1
[2017-11-21 01:49:23,036] client.Client<Client:98001>:OUTPUT: The operation result pair is= {'id': 0, 'client_id': 'c0', 'command_string': "put('movie','star')"} OK
[2017-11-21 01:49:23,037] client.Client<Client:98001>:OUTPUT: The result received is valid:(result, operation)= OK {'id': 0, 'client_id': 'c0', 'command_string': "put('movie','star')"}
[2017-11-21 01:49:23,037] client.Client<Client:98001>:OUTPUT: The following operation was created: {'id': 1, 'client_id': 'c0', 'command_string': "get('movie')"}
[2017-11-21 01:49:23,038] client.Client<Client:98001>:OUTPUT: return_value= star
[2017-11-21 01:49:23,038] replica.Replica<Replica:98002>:OUTPUT: Received result shuttle (source, client_id, operation_id, self_id)  r1 c0 0 r0
[2017-11-21 01:49:23,039] replica.Replica<Replica:98002>:OUTPUT: Checking for failure associated with trigger, application:  ('result_shuttle', 'pre')  in replica:  r0
[2017-11-21 01:49:23,039] replica.Replica<Replica:98002>:OUTPUT: No failure associated with trigger:  result_shuttle  found in replica:  r0
[2017-11-21 01:49:23,041] client.Client<Client:98001>:OUTPUT: A request for an operation was made {'content': {'is_retry': False, 'operation': {'id': 1, 'client_id': 'c0', 'command_string': "get('movie')"}}, 'signed_content': b"\x9f\x95\x1b{\xc5\x06\x1a\x0c\xc7x\xc5p=\x8c\xf0]\x99pw\x9ea\x91g\x06\xa2\xcd\xfe\xdd\x81\xd9\xdc&-\x83\xc7\x1amS\xac$o5\x19\xa7P\xbf\x9a%\xb3\xcc\x87\x04\x01\xd1K\xe6'N\xdc\xaf\xd6_\x91\x05\x80\x03}q\x00(X\x08\x00\x00\x00is_retryq\x01\x89X\t\x00\x00\x00operationq\x02}q\x03(X\x02\x00\x00\x00idq\x04K\x01X\t\x00\x00\x00client_idq\x05X\x02\x00\x00\x00c0q\x06X\x0e\x00\x00\x00command_stringq\x07X\x0c\x00\x00\x00get('movie')q\x08uu."}
[2017-11-21 01:49:23,053] replica.Replica<Replica:98002>:OUTPUT: Received request message (source, client_id, operation_id, self_id)  c0 c0 1 r0
[2017-11-21 01:49:23,054] replica.Replica<Replica:98002>:OUTPUT: Checking for failure associated with trigger, application:  ('client_request', 'pre')  in replica:  r0
[2017-11-21 01:49:23,054] replica.Replica<Replica:98002>:OUTPUT: No failure associated with trigger:  client_request  found in replica:  r0
[2017-11-21 01:49:23,057] replica.Replica<Replica:98002>:OUTPUT: Head is processing request (operation_id, client_id) (1, 'c0')
[2017-11-21 01:49:23,064] replica.Replica<Replica:98002>:OUTPUT: Checking for failure associated with trigger, application:  ('client_request', 'post')  in replica:  r0
[2017-11-21 01:49:23,065] replica.Replica<Replica:98002>:OUTPUT: No failure associated with trigger:  client_request  found in replica:  r0
[2017-11-21 01:49:23,068] replica.Replica<Replica:98003>:OUTPUT: Received shuttle (source, client_id, operation_id, self_id)  r0 c0 1 r1
[2017-11-21 01:49:23,068] replica.Replica<Replica:98003>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'pre')  in replica:  r1
[2017-11-21 01:49:23,069] replica.Replica<Replica:98003>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r1
[2017-11-21 01:49:23,083] replica.Replica<Replica:98003>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'post')  in replica:  r1
[2017-11-21 01:49:23,084] replica.Replica<Replica:98003>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r1
[2017-11-21 01:49:23,087] replica.Replica<Replica:98004>:OUTPUT: Received shuttle (source, client_id, operation_id, self_id)  r1 c0 1 r2
[2017-11-21 01:49:23,087] replica.Replica<Replica:98004>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'pre')  in replica:  r2
[2017-11-21 01:49:23,087] replica.Replica<Replica:98004>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r2
[2017-11-21 01:49:23,111] replica.Replica<Replica:98004>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'post')  in replica:  r2
[2017-11-21 01:49:23,112] replica.Replica<Replica:98004>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r2
[2017-11-21 01:49:23,115] replica.Replica<Replica:98003>:OUTPUT: Received result shuttle (source, client_id, operation_id, self_id)  r2 c0 1 r1
[2017-11-21 01:49:23,115] replica.Replica<Replica:98004>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'post')  in replica:  r2
[2017-11-21 01:49:23,115] replica.Replica<Replica:98003>:OUTPUT: Checking for failure associated with trigger, application:  ('result_shuttle', 'pre')  in replica:  r1
[2017-11-21 01:49:23,115] replica.Replica<Replica:98004>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r2
[2017-11-21 01:49:23,115] replica.Replica<Replica:98003>:OUTPUT: No failure associated with trigger:  result_shuttle  found in replica:  r1
[2017-11-21 01:49:23,118] client.Client<Client:98001>:OUTPUT: A result was received from replica= r2
[2017-11-21 01:49:23,127] replica.Replica<Replica:98003>:OUTPUT: Checking for failure associated with trigger, application:  ('result_shuttle', 'post')  in replica:  r1
[2017-11-21 01:49:23,128] replica.Replica<Replica:98003>:OUTPUT: No failure associated with trigger:  result_shuttle  found in replica:  r1
[2017-11-21 01:49:23,128] client.Client<Client:98001>:OUTPUT: The operation result pair is= {'id': 1, 'client_id': 'c0', 'command_string': "get('movie')"} star
[2017-11-21 01:49:23,129] client.Client<Client:98001>:OUTPUT: The result received is valid:(result, operation)= star {'id': 1, 'client_id': 'c0', 'command_string': "get('movie')"}
[2017-11-21 01:49:23,129] client.Client<Client:98001>:OUTPUT: The following operation was created: {'id': 2, 'client_id': 'c0', 'command_string': "append('movie', '123')"}
[2017-11-21 01:49:23,130] client.Client<Client:98001>:OUTPUT: return_value= OK
[2017-11-21 01:49:23,132] replica.Replica<Replica:98002>:OUTPUT: Received result shuttle (source, client_id, operation_id, self_id)  r1 c0 1 r0
[2017-11-21 01:49:23,133] replica.Replica<Replica:98002>:OUTPUT: Checking for failure associated with trigger, application:  ('result_shuttle', 'pre')  in replica:  r0
[2017-11-21 01:49:23,133] replica.Replica<Replica:98002>:OUTPUT: No failure associated with trigger:  result_shuttle  found in replica:  r0
[2017-11-21 01:49:23,134] client.Client<Client:98001>:OUTPUT: A request for an operation was made {'content': {'is_retry': False, 'operation': {'id': 2, 'client_id': 'c0', 'command_string': "append('movie', '123')"}}, 'signed_content': b"\x9e\\\xafwU\xd6\xf48t,\\h5c\x16'\x07\x1bg\x89\xac\x8b\xb6\xed\xf9\xe9\xfa\xc4\x8dz!p\xd9/\xe9)\xa3c\xa1\xd7\xb5\n\x83)D\x892\x1cX\x92\xf0\x80\x02\x18\x9f 3\x96\nQ\xa5Xa\x01\x80\x03}q\x00(X\x08\x00\x00\x00is_retryq\x01\x89X\t\x00\x00\x00operationq\x02}q\x03(X\x02\x00\x00\x00idq\x04K\x02X\t\x00\x00\x00client_idq\x05X\x02\x00\x00\x00c0q\x06X\x0e\x00\x00\x00command_stringq\x07X\x16\x00\x00\x00append('movie', '123')q\x08uu."}
[2017-11-21 01:49:23,147] replica.Replica<Replica:98002>:OUTPUT: Received request message (source, client_id, operation_id, self_id)  c0 c0 2 r0
[2017-11-21 01:49:23,147] replica.Replica<Replica:98002>:OUTPUT: Checking for failure associated with trigger, application:  ('client_request', 'pre')  in replica:  r0
[2017-11-21 01:49:23,147] replica.Replica<Replica:98002>:OUTPUT: No failure associated with trigger:  client_request  found in replica:  r0
[2017-11-21 01:49:23,152] replica.Replica<Replica:98002>:OUTPUT: Head is processing request (operation_id, client_id) (2, 'c0')
[2017-11-21 01:49:23,159] replica.Replica<Replica:98002>:OUTPUT: Checking for failure associated with trigger, application:  ('client_request', 'post')  in replica:  r0
[2017-11-21 01:49:23,160] replica.Replica<Replica:98002>:OUTPUT: No failure associated with trigger:  client_request  found in replica:  r0
[2017-11-21 01:49:23,162] replica.Replica<Replica:98003>:OUTPUT: Received shuttle (source, client_id, operation_id, self_id)  r0 c0 2 r1
[2017-11-21 01:49:23,162] replica.Replica<Replica:98003>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'pre')  in replica:  r1
[2017-11-21 01:49:23,163] replica.Replica<Replica:98003>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r1
[2017-11-21 01:49:23,179] replica.Replica<Replica:98003>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'post')  in replica:  r1
[2017-11-21 01:49:23,180] replica.Replica<Replica:98003>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r1
[2017-11-21 01:49:23,183] replica.Replica<Replica:98004>:OUTPUT: Received shuttle (source, client_id, operation_id, self_id)  r1 c0 2 r2
[2017-11-21 01:49:23,183] replica.Replica<Replica:98004>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'pre')  in replica:  r2
[2017-11-21 01:49:23,184] replica.Replica<Replica:98004>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r2
[2017-11-21 01:49:23,205] replica.Replica<Replica:98004>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'post')  in replica:  r2
[2017-11-21 01:49:23,207] replica.Replica<Replica:98004>:OUTPUT: Failure invalid_result_sig applied at replica:  r2
[2017-11-21 01:49:23,211] replica.Replica<Replica:98003>:OUTPUT: Received result shuttle (source, client_id, operation_id, self_id)  r2 c0 2 r1
[2017-11-21 01:49:23,211] replica.Replica<Replica:98004>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'post')  in replica:  r2
[2017-11-21 01:49:23,211] replica.Replica<Replica:98003>:OUTPUT: Checking for failure associated with trigger, application:  ('result_shuttle', 'pre')  in replica:  r1
[2017-11-21 01:49:23,211] replica.Replica<Replica:98003>:OUTPUT: No failure associated with trigger:  result_shuttle  found in replica:  r1
[2017-11-21 01:49:23,213] replica.Replica<Replica:98004>:OUTPUT: Failure invalid_result_sig applied at replica:  r2
[2017-11-21 01:49:23,215] replica.Replica<Replica:98003>:OUTPUT: A signature mismatch occurred from source: r2
[2017-11-21 01:49:23,216] client.Client<Client:98001>:OUTPUT: A result was received from replica= r2
[2017-11-21 01:49:23,231] client.Client<Client:98001>:OUTPUT: The operation result pair is= {'id': 2, 'client_id': 'c0', 'command_string': "append('movie', '123')"} OK
[2017-11-21 01:49:23,231] client.Client<Client:98001>:OUTPUT: The result received is valid:(result, operation)= OK {'id': 2, 'client_id': 'c0', 'command_string': "append('movie', '123')"}
[2017-11-21 01:49:23,232] client.Client<Client:98001>:OUTPUT: The following operation was created: {'id': 3, 'client_id': 'c0', 'command_string': "get('movie')"}
[2017-11-21 01:49:23,232] client.Client<Client:98001>:OUTPUT: return_value= star123
[2017-11-21 01:49:23,234] client.Client<Client:98001>:OUTPUT: A request for an operation was made {'content': {'is_retry': False, 'operation': {'id': 3, 'client_id': 'c0', 'command_string': "get('movie')"}}, 'signed_content': b"\xea/(,\xf3\xdd\xd2\xe83\xb568\x80rE?I\xcd\xe4m4\xd3\xef\x9a\xb1\x18\xaa\x1c\xa8\xc9\x1b6\xba\xefB$X\x07\xf1\xb8[\x91\x99\xd6\xd7X\x10~\xdd\xc1\xf3\xd15\x0e\x02\x0cE#\x9e\x00\x8d#\xbe\x06\x80\x03}q\x00(X\x08\x00\x00\x00is_retryq\x01\x89X\t\x00\x00\x00operationq\x02}q\x03(X\x02\x00\x00\x00idq\x04K\x03X\t\x00\x00\x00client_idq\x05X\x02\x00\x00\x00c0q\x06X\x0e\x00\x00\x00command_stringq\x07X\x0c\x00\x00\x00get('movie')q\x08uu."}
[2017-11-21 01:49:23,235] replica.Replica<Replica:98002>:OUTPUT: Received request message (source, client_id, operation_id, self_id)  c0 c0 3 r0
[2017-11-21 01:49:23,235] replica.Replica<Replica:98002>:OUTPUT: Checking for failure associated with trigger, application:  ('client_request', 'pre')  in replica:  r0
[2017-11-21 01:49:23,236] replica.Replica<Replica:98002>:OUTPUT: No failure associated with trigger:  client_request  found in replica:  r0
[2017-11-21 01:49:23,238] replica.Replica<Replica:98002>:OUTPUT: Head is processing request (operation_id, client_id) (3, 'c0')
[2017-11-21 01:49:23,243] replica.Replica<Replica:98002>:OUTPUT: Checking for failure associated with trigger, application:  ('client_request', 'post')  in replica:  r0
[2017-11-21 01:49:23,244] replica.Replica<Replica:98002>:OUTPUT: No failure associated with trigger:  client_request  found in replica:  r0
[2017-11-21 01:49:23,246] replica.Replica<Replica:98003>:OUTPUT: Received shuttle (source, client_id, operation_id, self_id)  r0 c0 3 r1
[2017-11-21 01:49:23,247] replica.Replica<Replica:98003>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'pre')  in replica:  r1
[2017-11-21 01:49:23,247] replica.Replica<Replica:98003>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r1
[2017-11-21 01:49:23,262] replica.Replica<Replica:98003>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'post')  in replica:  r1
[2017-11-21 01:49:23,262] replica.Replica<Replica:98003>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r1
[2017-11-21 01:49:23,265] replica.Replica<Replica:98004>:OUTPUT: Received shuttle (source, client_id, operation_id, self_id)  r1 c0 3 r2
[2017-11-21 01:49:23,265] replica.Replica<Replica:98004>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'pre')  in replica:  r2
[2017-11-21 01:49:23,265] replica.Replica<Replica:98004>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r2
[2017-11-21 01:49:23,289] replica.Replica<Replica:98004>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'post')  in replica:  r2
[2017-11-21 01:49:23,289] replica.Replica<Replica:98004>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r2
[2017-11-21 01:49:23,293] replica.Replica<Replica:98003>:OUTPUT: Received result shuttle (source, client_id, operation_id, self_id)  r2 c0 3 r1
[2017-11-21 01:49:23,293] replica.Replica<Replica:98003>:OUTPUT: Checking for failure associated with trigger, application:  ('result_shuttle', 'pre')  in replica:  r1
[2017-11-21 01:49:23,293] replica.Replica<Replica:98004>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'post')  in replica:  r2
[2017-11-21 01:49:23,294] replica.Replica<Replica:98003>:OUTPUT: No failure associated with trigger:  result_shuttle  found in replica:  r1
[2017-11-21 01:49:23,294] replica.Replica<Replica:98004>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r2
[2017-11-21 01:49:23,297] client.Client<Client:98001>:OUTPUT: A result was received from replica= r2
[2017-11-21 01:49:23,311] replica.Replica<Replica:98003>:OUTPUT: Checking for failure associated with trigger, application:  ('result_shuttle', 'post')  in replica:  r1
[2017-11-21 01:49:23,311] replica.Replica<Replica:98003>:OUTPUT: No failure associated with trigger:  result_shuttle  found in replica:  r1
[2017-11-21 01:49:23,312] client.Client<Client:98001>:OUTPUT: The operation result pair is= {'id': 3, 'client_id': 'c0', 'command_string': "get('movie')"} star123
[2017-11-21 01:49:23,313] client.Client<Client:98001>:OUTPUT: The result received is valid:(result, operation)= star123 {'id': 3, 'client_id': 'c0', 'command_string': "get('movie')"}
[2017-11-21 01:49:23,313] client.Client<Client:98001>:OUTPUT: The following operation was created: {'id': 4, 'client_id': 'c0', 'command_string': "get('movie')"}
[2017-11-21 01:49:23,313] client.Client<Client:98001>:OUTPUT: return_value= star123
[2017-11-21 01:49:23,315] replica.Replica<Replica:98002>:OUTPUT: Received result shuttle (source, client_id, operation_id, self_id)  r1 c0 3 r0
[2017-11-21 01:49:23,315] replica.Replica<Replica:98002>:OUTPUT: Checking for failure associated with trigger, application:  ('result_shuttle', 'pre')  in replica:  r0
[2017-11-21 01:49:23,316] replica.Replica<Replica:98002>:OUTPUT: No failure associated with trigger:  result_shuttle  found in replica:  r0
[2017-11-21 01:49:23,316] client.Client<Client:98001>:OUTPUT: A request for an operation was made {'content': {'is_retry': False, 'operation': {'id': 4, 'client_id': 'c0', 'command_string': "get('movie')"}}, 'signed_content': b'\xa01\x95"\xaaka7M(x\x92u\xc1\xa3\xcbV\xdee\r^\xa9Jp+y\xb0\xdb^\xafm:\xc6\xb9\x8c\xb9\xe2_\xc7*=\xb4\xdfs\rF\xb0\x90\xd97\x1fT\'\xb1cR\x8a\x99\xe2\x00`^y\x03\x80\x03}q\x00(X\x08\x00\x00\x00is_retryq\x01\x89X\t\x00\x00\x00operationq\x02}q\x03(X\x02\x00\x00\x00idq\x04K\x04X\t\x00\x00\x00client_idq\x05X\x02\x00\x00\x00c0q\x06X\x0e\x00\x00\x00command_stringq\x07X\x0c\x00\x00\x00get(\'movie\')q\x08uu.'}
[2017-11-21 01:49:23,329] replica.Replica<Replica:98002>:OUTPUT: Received request message (source, client_id, operation_id, self_id)  c0 c0 4 r0
[2017-11-21 01:49:23,329] replica.Replica<Replica:98002>:OUTPUT: Checking for failure associated with trigger, application:  ('client_request', 'pre')  in replica:  r0
[2017-11-21 01:49:23,330] replica.Replica<Replica:98002>:OUTPUT: No failure associated with trigger:  client_request  found in replica:  r0
[2017-11-21 01:49:23,332] replica.Replica<Replica:98002>:OUTPUT: Head is processing request (operation_id, client_id) (4, 'c0')
[2017-11-21 01:49:23,338] replica.Replica<Replica:98002>:OUTPUT: Checking for failure associated with trigger, application:  ('client_request', 'post')  in replica:  r0
[2017-11-21 01:49:23,338] replica.Replica<Replica:98002>:OUTPUT: No failure associated with trigger:  client_request  found in replica:  r0
[2017-11-21 01:49:23,342] replica.Replica<Replica:98003>:OUTPUT: Received shuttle (source, client_id, operation_id, self_id)  r0 c0 4 r1
[2017-11-21 01:49:23,342] replica.Replica<Replica:98003>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'pre')  in replica:  r1
[2017-11-21 01:49:23,343] replica.Replica<Replica:98003>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r1
[2017-11-21 01:49:23,358] replica.Replica<Replica:98003>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'post')  in replica:  r1
[2017-11-21 01:49:23,359] replica.Replica<Replica:98003>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r1
[2017-11-21 01:49:23,363] replica.Replica<Replica:98004>:OUTPUT: Received shuttle (source, client_id, operation_id, self_id)  r1 c0 4 r2
[2017-11-21 01:49:23,364] replica.Replica<Replica:98004>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'pre')  in replica:  r2
[2017-11-21 01:49:23,365] replica.Replica<Replica:98004>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r2
[2017-11-21 01:49:23,395] replica.Replica<Replica:98004>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'post')  in replica:  r2
[2017-11-21 01:49:23,395] replica.Replica<Replica:98004>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r2
[2017-11-21 01:49:23,398] replica.Replica<Replica:98003>:OUTPUT: Received result shuttle (source, client_id, operation_id, self_id)  r2 c0 4 r1
[2017-11-21 01:49:23,398] replica.Replica<Replica:98003>:OUTPUT: Checking for failure associated with trigger, application:  ('result_shuttle', 'pre')  in replica:  r1
[2017-11-21 01:49:23,398] replica.Replica<Replica:98004>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'post')  in replica:  r2
[2017-11-21 01:49:23,399] replica.Replica<Replica:98003>:OUTPUT: No failure associated with trigger:  result_shuttle  found in replica:  r1
[2017-11-21 01:49:23,399] replica.Replica<Replica:98004>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r2
[2017-11-21 01:49:23,401] client.Client<Client:98001>:OUTPUT: A result was received from replica= r2
[2017-11-21 01:49:23,411] replica.Replica<Replica:98003>:OUTPUT: Checking for failure associated with trigger, application:  ('result_shuttle', 'post')  in replica:  r1
[2017-11-21 01:49:23,411] replica.Replica<Replica:98003>:OUTPUT: No failure associated with trigger:  result_shuttle  found in replica:  r1
[2017-11-21 01:49:23,411] client.Client<Client:98001>:OUTPUT: The operation result pair is= {'id': 4, 'client_id': 'c0', 'command_string': "get('movie')"} star123
[2017-11-21 01:49:23,412] client.Client<Client:98001>:OUTPUT: The result received is valid:(result, operation)= star123 {'id': 4, 'client_id': 'c0', 'command_string': "get('movie')"}
[2017-11-21 01:49:23,412] client.Client<Client:98001>:OUTPUT: The following operation was created: {'id': 5, 'client_id': 'c0', 'command_string': "get('movie')"}
[2017-11-21 01:49:23,413] client.Client<Client:98001>:OUTPUT: return_value= star123
[2017-11-21 01:49:23,415] replica.Replica<Replica:98002>:OUTPUT: Received result shuttle (source, client_id, operation_id, self_id)  r1 c0 4 r0
[2017-11-21 01:49:23,416] replica.Replica<Replica:98002>:OUTPUT: Checking for failure associated with trigger, application:  ('result_shuttle', 'pre')  in replica:  r0
[2017-11-21 01:49:23,416] replica.Replica<Replica:98002>:OUTPUT: No failure associated with trigger:  result_shuttle  found in replica:  r0
[2017-11-21 01:49:23,416] client.Client<Client:98001>:OUTPUT: A request for an operation was made {'content': {'is_retry': False, 'operation': {'id': 5, 'client_id': 'c0', 'command_string': "get('movie')"}}, 'signed_content': b"\x01vz\xf0yQu\xea\x02\xfa]Cg.\xc4\xab+\x8a\x8bU\xa2\xa8m\xd8\xec\x12\xa0oV\x97W\xca\xc8\xe2\xf3\\\xd0\xbf\xa6\xbabF\xfd\x81\x03\x82\xe19HN(\x17N\xa2\xbc\xe4\xf5\xf3\x90\x98\xd7\x1d\xd9\x0f\x80\x03}q\x00(X\x08\x00\x00\x00is_retryq\x01\x89X\t\x00\x00\x00operationq\x02}q\x03(X\x02\x00\x00\x00idq\x04K\x05X\t\x00\x00\x00client_idq\x05X\x02\x00\x00\x00c0q\x06X\x0e\x00\x00\x00command_stringq\x07X\x0c\x00\x00\x00get('movie')q\x08uu."}
[2017-11-21 01:49:23,431] replica.Replica<Replica:98002>:OUTPUT: Received request message (source, client_id, operation_id, self_id)  c0 c0 5 r0
[2017-11-21 01:49:23,432] replica.Replica<Replica:98002>:OUTPUT: Checking for failure associated with trigger, application:  ('client_request', 'pre')  in replica:  r0
[2017-11-21 01:49:23,432] replica.Replica<Replica:98002>:OUTPUT: No failure associated with trigger:  client_request  found in replica:  r0
[2017-11-21 01:49:23,435] replica.Replica<Replica:98002>:OUTPUT: Head is processing request (operation_id, client_id) (5, 'c0')
[2017-11-21 01:49:23,442] replica.Replica<Replica:98002>:OUTPUT: Checking for failure associated with trigger, application:  ('client_request', 'post')  in replica:  r0
[2017-11-21 01:49:23,443] replica.Replica<Replica:98002>:OUTPUT: No failure associated with trigger:  client_request  found in replica:  r0
[2017-11-21 01:49:23,445] replica.Replica<Replica:98003>:OUTPUT: Received shuttle (source, client_id, operation_id, self_id)  r0 c0 5 r1
[2017-11-21 01:49:23,446] replica.Replica<Replica:98003>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'pre')  in replica:  r1
[2017-11-21 01:49:23,446] replica.Replica<Replica:98003>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r1
[2017-11-21 01:49:23,460] replica.Replica<Replica:98003>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'post')  in replica:  r1
[2017-11-21 01:49:23,461] replica.Replica<Replica:98003>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r1
[2017-11-21 01:49:23,463] replica.Replica<Replica:98004>:OUTPUT: Received shuttle (source, client_id, operation_id, self_id)  r1 c0 5 r2
[2017-11-21 01:49:23,463] replica.Replica<Replica:98004>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'pre')  in replica:  r2
[2017-11-21 01:49:23,464] replica.Replica<Replica:98004>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r2
[2017-11-21 01:49:23,487] replica.Replica<Replica:98004>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'post')  in replica:  r2
[2017-11-21 01:49:23,487] replica.Replica<Replica:98004>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r2
[2017-11-21 01:49:23,490] replica.Replica<Replica:98003>:OUTPUT: Received result shuttle (source, client_id, operation_id, self_id)  r2 c0 5 r1
[2017-11-21 01:49:23,491] replica.Replica<Replica:98004>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'post')  in replica:  r2
[2017-11-21 01:49:23,491] replica.Replica<Replica:98003>:OUTPUT: Checking for failure associated with trigger, application:  ('result_shuttle', 'pre')  in replica:  r1
[2017-11-21 01:49:23,491] replica.Replica<Replica:98004>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r2
[2017-11-21 01:49:23,491] replica.Replica<Replica:98003>:OUTPUT: No failure associated with trigger:  result_shuttle  found in replica:  r1
[2017-11-21 01:49:23,494] client.Client<Client:98001>:OUTPUT: A result was received from replica= r2
[2017-11-21 01:49:23,509] replica.Replica<Replica:98003>:OUTPUT: Checking for failure associated with trigger, application:  ('result_shuttle', 'post')  in replica:  r1
[2017-11-21 01:49:23,510] client.Client<Client:98001>:OUTPUT: The operation result pair is= {'id': 5, 'client_id': 'c0', 'command_string': "get('movie')"} star123
[2017-11-21 01:49:23,511] replica.Replica<Replica:98003>:OUTPUT: No failure associated with trigger:  result_shuttle  found in replica:  r1
[2017-11-21 01:49:23,511] client.Client<Client:98001>:OUTPUT: The result received is valid:(result, operation)= star123 {'id': 5, 'client_id': 'c0', 'command_string': "get('movie')"}
[2017-11-21 01:49:23,512] client.Client<Client:98001>:OUTPUT: The following operation was created: {'id': 6, 'client_id': 'c0', 'command_string': "get('movie')"}
[2017-11-21 01:49:23,512] client.Client<Client:98001>:OUTPUT: return_value= star123
[2017-11-21 01:49:23,514] replica.Replica<Replica:98002>:OUTPUT: Received result shuttle (source, client_id, operation_id, self_id)  r1 c0 5 r0
[2017-11-21 01:49:23,514] client.Client<Client:98001>:OUTPUT: A request for an operation was made {'content': {'is_retry': False, 'operation': {'id': 6, 'client_id': 'c0', 'command_string': "get('movie')"}}, 'signed_content': b"\xc9\xb69*e\x80O'\xd0\x9a\x9d\x886\xbd\xb8andy\x13\x90\x01MRa\x0e\xa7|3*Kh\x08L6MV\xe9\xdf\xb9b\xfe\xb7\xec\x82^\x14\xff\xe30\xc3\x8fV\xf6\xeayCp\xd3y\xc3\xf1\xa6\x06\x80\x03}q\x00(X\x08\x00\x00\x00is_retryq\x01\x89X\t\x00\x00\x00operationq\x02}q\x03(X\x02\x00\x00\x00idq\x04K\x06X\t\x00\x00\x00client_idq\x05X\x02\x00\x00\x00c0q\x06X\x0e\x00\x00\x00command_stringq\x07X\x0c\x00\x00\x00get('movie')q\x08uu."}
[2017-11-21 01:49:23,515] replica.Replica<Replica:98002>:OUTPUT: Checking for failure associated with trigger, application:  ('result_shuttle', 'pre')  in replica:  r0
[2017-11-21 01:49:23,515] replica.Replica<Replica:98002>:OUTPUT: No failure associated with trigger:  result_shuttle  found in replica:  r0
[2017-11-21 01:49:23,529] replica.Replica<Replica:98002>:OUTPUT: Received request message (source, client_id, operation_id, self_id)  c0 c0 6 r0
[2017-11-21 01:49:23,530] replica.Replica<Replica:98002>:OUTPUT: Checking for failure associated with trigger, application:  ('client_request', 'pre')  in replica:  r0
[2017-11-21 01:49:23,530] replica.Replica<Replica:98002>:OUTPUT: No failure associated with trigger:  client_request  found in replica:  r0
[2017-11-21 01:49:23,533] replica.Replica<Replica:98002>:OUTPUT: Head is processing request (operation_id, client_id) (6, 'c0')
[2017-11-21 01:49:23,540] replica.Replica<Replica:98002>:OUTPUT: Checking for failure associated with trigger, application:  ('client_request', 'post')  in replica:  r0
[2017-11-21 01:49:23,541] replica.Replica<Replica:98002>:OUTPUT: No failure associated with trigger:  client_request  found in replica:  r0
[2017-11-21 01:49:23,545] replica.Replica<Replica:98003>:OUTPUT: Received shuttle (source, client_id, operation_id, self_id)  r0 c0 6 r1
[2017-11-21 01:49:23,545] replica.Replica<Replica:98003>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'pre')  in replica:  r1
[2017-11-21 01:49:23,545] replica.Replica<Replica:98003>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r1
[2017-11-21 01:49:23,559] replica.Replica<Replica:98003>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'post')  in replica:  r1
[2017-11-21 01:49:23,560] replica.Replica<Replica:98003>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r1
[2017-11-21 01:49:23,563] replica.Replica<Replica:98004>:OUTPUT: Received shuttle (source, client_id, operation_id, self_id)  r1 c0 6 r2
[2017-11-21 01:49:23,564] replica.Replica<Replica:98004>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'pre')  in replica:  r2
[2017-11-21 01:49:23,564] replica.Replica<Replica:98004>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r2
[2017-11-21 01:49:23,590] replica.Replica<Replica:98004>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'post')  in replica:  r2
[2017-11-21 01:49:23,590] replica.Replica<Replica:98004>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r2
[2017-11-21 01:49:23,594] replica.Replica<Replica:98003>:OUTPUT: Received result shuttle (source, client_id, operation_id, self_id)  r2 c0 6 r1
[2017-11-21 01:49:23,594] replica.Replica<Replica:98004>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'post')  in replica:  r2
[2017-11-21 01:49:23,595] replica.Replica<Replica:98003>:OUTPUT: Checking for failure associated with trigger, application:  ('result_shuttle', 'pre')  in replica:  r1
[2017-11-21 01:49:23,595] replica.Replica<Replica:98004>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r2
[2017-11-21 01:49:23,595] replica.Replica<Replica:98003>:OUTPUT: No failure associated with trigger:  result_shuttle  found in replica:  r1
[2017-11-21 01:49:23,597] client.Client<Client:98001>:OUTPUT: A result was received from replica= r2
[2017-11-21 01:49:23,612] client.Client<Client:98001>:OUTPUT: The operation result pair is= {'id': 6, 'client_id': 'c0', 'command_string': "get('movie')"} star123
[2017-11-21 01:49:23,612] client.Client<Client:98001>:OUTPUT: The result received is valid:(result, operation)= star123 {'id': 6, 'client_id': 'c0', 'command_string': "get('movie')"}
[2017-11-21 01:49:23,612] replica.Replica<Replica:98003>:OUTPUT: Checking for failure associated with trigger, application:  ('result_shuttle', 'post')  in replica:  r1
[2017-11-21 01:49:23,613] client.Client<Client:98001>:OUTPUT: The following operation was created: {'id': 7, 'client_id': 'c0', 'command_string': "get('movie')"}
[2017-11-21 01:49:23,613] replica.Replica<Replica:98003>:OUTPUT: No failure associated with trigger:  result_shuttle  found in replica:  r1
[2017-11-21 01:49:23,613] client.Client<Client:98001>:OUTPUT: return_value= star123
[2017-11-21 01:49:23,616] client.Client<Client:98001>:OUTPUT: A request for an operation was made {'content': {'is_retry': False, 'operation': {'id': 7, 'client_id': 'c0', 'command_string': "get('movie')"}}, 'signed_content': b"`\x93\xaa\xc0/cr\xaa}\x92\x8e\xfdX\xcd\xdd\xc7\x96\xcc\xa0\xea\x05\x04\x055\xe5_\xb4\xf1\x06\x8b\xec\xady~G\xc6[r9g\xa8Pn/'\xc4\xe2\xf2\x9dd_(\xfa\x0ec\xe51TD[\t\xb6\xfe\x0c\x80\x03}q\x00(X\x08\x00\x00\x00is_retryq\x01\x89X\t\x00\x00\x00operationq\x02}q\x03(X\x02\x00\x00\x00idq\x04K\x07X\t\x00\x00\x00client_idq\x05X\x02\x00\x00\x00c0q\x06X\x0e\x00\x00\x00command_stringq\x07X\x0c\x00\x00\x00get('movie')q\x08uu."}
[2017-11-21 01:49:23,616] replica.Replica<Replica:98002>:OUTPUT: Received result shuttle (source, client_id, operation_id, self_id)  r1 c0 6 r0
[2017-11-21 01:49:23,617] replica.Replica<Replica:98002>:OUTPUT: Checking for failure associated with trigger, application:  ('result_shuttle', 'pre')  in replica:  r0
[2017-11-21 01:49:23,617] replica.Replica<Replica:98002>:OUTPUT: No failure associated with trigger:  result_shuttle  found in replica:  r0
[2017-11-21 01:49:23,629] replica.Replica<Replica:98002>:OUTPUT: Received request message (source, client_id, operation_id, self_id)  c0 c0 7 r0
[2017-11-21 01:49:23,630] replica.Replica<Replica:98002>:OUTPUT: Checking for failure associated with trigger, application:  ('client_request', 'pre')  in replica:  r0
[2017-11-21 01:49:23,630] replica.Replica<Replica:98002>:OUTPUT: No failure associated with trigger:  client_request  found in replica:  r0
[2017-11-21 01:49:23,633] replica.Replica<Replica:98002>:OUTPUT: Head is processing request (operation_id, client_id) (7, 'c0')
[2017-11-21 01:49:23,640] replica.Replica<Replica:98002>:OUTPUT: Checking for failure associated with trigger, application:  ('client_request', 'post')  in replica:  r0
[2017-11-21 01:49:23,641] replica.Replica<Replica:98002>:OUTPUT: No failure associated with trigger:  client_request  found in replica:  r0
[2017-11-21 01:49:23,645] replica.Replica<Replica:98003>:OUTPUT: Received shuttle (source, client_id, operation_id, self_id)  r0 c0 7 r1
[2017-11-21 01:49:23,645] replica.Replica<Replica:98003>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'pre')  in replica:  r1
[2017-11-21 01:49:23,645] replica.Replica<Replica:98003>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r1
[2017-11-21 01:49:23,659] replica.Replica<Replica:98003>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'post')  in replica:  r1
[2017-11-21 01:49:23,660] replica.Replica<Replica:98003>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r1
[2017-11-21 01:49:23,663] replica.Replica<Replica:98004>:OUTPUT: Received shuttle (source, client_id, operation_id, self_id)  r1 c0 7 r2
[2017-11-21 01:49:23,663] replica.Replica<Replica:98004>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'pre')  in replica:  r2
[2017-11-21 01:49:23,664] replica.Replica<Replica:98004>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r2
[2017-11-21 01:49:23,690] replica.Replica<Replica:98004>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'post')  in replica:  r2
[2017-11-21 01:49:23,691] replica.Replica<Replica:98004>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r2
[2017-11-21 01:49:23,694] replica.Replica<Replica:98003>:OUTPUT: Received result shuttle (source, client_id, operation_id, self_id)  r2 c0 7 r1
[2017-11-21 01:49:23,694] replica.Replica<Replica:98004>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'post')  in replica:  r2
[2017-11-21 01:49:23,695] replica.Replica<Replica:98003>:OUTPUT: Checking for failure associated with trigger, application:  ('result_shuttle', 'pre')  in replica:  r1
[2017-11-21 01:49:23,695] replica.Replica<Replica:98004>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r2
[2017-11-21 01:49:23,695] replica.Replica<Replica:98003>:OUTPUT: No failure associated with trigger:  result_shuttle  found in replica:  r1
[2017-11-21 01:49:23,698] client.Client<Client:98001>:OUTPUT: A result was received from replica= r2
[2017-11-21 01:49:23,710] client.Client<Client:98001>:OUTPUT: The operation result pair is= {'id': 7, 'client_id': 'c0', 'command_string': "get('movie')"} star123
[2017-11-21 01:49:23,711] client.Client<Client:98001>:OUTPUT: The result received is valid:(result, operation)= star123 {'id': 7, 'client_id': 'c0', 'command_string': "get('movie')"}
[2017-11-21 01:49:23,712] client.Client<Client:98001>:OUTPUT: The following operation was created: {'id': 8, 'client_id': 'c0', 'command_string': "get('movie')"}
[2017-11-21 01:49:23,712] client.Client<Client:98001>:OUTPUT: return_value= star123
[2017-11-21 01:49:23,713] replica.Replica<Replica:98003>:OUTPUT: Checking for failure associated with trigger, application:  ('result_shuttle', 'post')  in replica:  r1
[2017-11-21 01:49:23,713] replica.Replica<Replica:98003>:OUTPUT: No failure associated with trigger:  result_shuttle  found in replica:  r1
[2017-11-21 01:49:23,715] client.Client<Client:98001>:OUTPUT: A request for an operation was made {'content': {'is_retry': False, 'operation': {'id': 8, 'client_id': 'c0', 'command_string': "get('movie')"}}, 'signed_content': b"\xfd\xb7\xcc\xf2\xd2\xb72p&D\xfb\x07\xb9\xdf\x89\xc6\xfa&0\xba\x92\xcb\xbd\xdc\xd3\x0e\x1f\xb6fG\xddF\xae|\x15\xa7\x9b\x8c\xc2\xeb\xb8%\xc0\x81\xd0\x002VL\x94t\x01\xf3\x93!\xae\xc6\xe2\xd6..\x95Z\r\x80\x03}q\x00(X\x08\x00\x00\x00is_retryq\x01\x89X\t\x00\x00\x00operationq\x02}q\x03(X\x02\x00\x00\x00idq\x04K\x08X\t\x00\x00\x00client_idq\x05X\x02\x00\x00\x00c0q\x06X\x0e\x00\x00\x00command_stringq\x07X\x0c\x00\x00\x00get('movie')q\x08uu."}
[2017-11-21 01:49:23,716] replica.Replica<Replica:98002>:OUTPUT: Received result shuttle (source, client_id, operation_id, self_id)  r1 c0 7 r0
[2017-11-21 01:49:23,717] replica.Replica<Replica:98002>:OUTPUT: Checking for failure associated with trigger, application:  ('result_shuttle', 'pre')  in replica:  r0
[2017-11-21 01:49:23,718] replica.Replica<Replica:98002>:OUTPUT: No failure associated with trigger:  result_shuttle  found in replica:  r0
[2017-11-21 01:49:23,733] replica.Replica<Replica:98002>:OUTPUT: Received request message (source, client_id, operation_id, self_id)  c0 c0 8 r0
[2017-11-21 01:49:23,734] replica.Replica<Replica:98002>:OUTPUT: Checking for failure associated with trigger, application:  ('client_request', 'pre')  in replica:  r0
[2017-11-21 01:49:23,734] replica.Replica<Replica:98002>:OUTPUT: No failure associated with trigger:  client_request  found in replica:  r0
[2017-11-21 01:49:23,737] replica.Replica<Replica:98002>:OUTPUT: Head is processing request (operation_id, client_id) (8, 'c0')
[2017-11-21 01:49:23,741] replica.Replica<Replica:98002>:OUTPUT: Checking for failure associated with trigger, application:  ('client_request', 'post')  in replica:  r0
[2017-11-21 01:49:23,741] replica.Replica<Replica:98002>:OUTPUT: No failure associated with trigger:  client_request  found in replica:  r0
[2017-11-21 01:49:23,744] replica.Replica<Replica:98003>:OUTPUT: Received shuttle (source, client_id, operation_id, self_id)  r0 c0 8 r1
[2017-11-21 01:49:23,745] replica.Replica<Replica:98003>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'pre')  in replica:  r1
[2017-11-21 01:49:23,745] replica.Replica<Replica:98003>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r1
[2017-11-21 01:49:23,757] replica.Replica<Replica:98003>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'post')  in replica:  r1
[2017-11-21 01:49:23,757] replica.Replica<Replica:98003>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r1
[2017-11-21 01:49:23,761] replica.Replica<Replica:98004>:OUTPUT: Received shuttle (source, client_id, operation_id, self_id)  r1 c0 8 r2
[2017-11-21 01:49:23,761] replica.Replica<Replica:98004>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'pre')  in replica:  r2
[2017-11-21 01:49:23,762] replica.Replica<Replica:98004>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r2
[2017-11-21 01:49:23,786] replica.Replica<Replica:98004>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'post')  in replica:  r2
[2017-11-21 01:49:23,786] replica.Replica<Replica:98004>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r2
[2017-11-21 01:49:23,789] replica.Replica<Replica:98003>:OUTPUT: Received result shuttle (source, client_id, operation_id, self_id)  r2 c0 8 r1
[2017-11-21 01:49:23,790] replica.Replica<Replica:98004>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'post')  in replica:  r2
[2017-11-21 01:49:23,790] replica.Replica<Replica:98003>:OUTPUT: Checking for failure associated with trigger, application:  ('result_shuttle', 'pre')  in replica:  r1
[2017-11-21 01:49:23,790] replica.Replica<Replica:98004>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r2
[2017-11-21 01:49:23,790] replica.Replica<Replica:98003>:OUTPUT: No failure associated with trigger:  result_shuttle  found in replica:  r1
[2017-11-21 01:49:23,792] client.Client<Client:98001>:OUTPUT: A result was received from replica= r2
[2017-11-21 01:49:23,806] client.Client<Client:98001>:OUTPUT: The operation result pair is= {'id': 8, 'client_id': 'c0', 'command_string': "get('movie')"} star123
[2017-11-21 01:49:23,807] client.Client<Client:98001>:OUTPUT: The result received is valid:(result, operation)= star123 {'id': 8, 'client_id': 'c0', 'command_string': "get('movie')"}
[2017-11-21 01:49:23,807] replica.Replica<Replica:98003>:OUTPUT: Checking for failure associated with trigger, application:  ('result_shuttle', 'post')  in replica:  r1
[2017-11-21 01:49:23,807] client.Client<Client:98001>:OUTPUT: The following operation was created: {'id': 9, 'client_id': 'c0', 'command_string': "get('movie')"}
[2017-11-21 01:49:23,807] replica.Replica<Replica:98003>:OUTPUT: No failure associated with trigger:  result_shuttle  found in replica:  r1
[2017-11-21 01:49:23,808] client.Client<Client:98001>:OUTPUT: return_value= star123
[2017-11-21 01:49:23,812] client.Client<Client:98001>:OUTPUT: A request for an operation was made {'content': {'is_retry': False, 'operation': {'id': 9, 'client_id': 'c0', 'command_string': "get('movie')"}}, 'signed_content': b"7K\xa3\x0e\xecA)m0djb\xbd\x94\xaf\xee\x93y\xc5dO\xd3\xde\x1e\x8f\xf7>E\xda\x96.\x19D3#-\x1d?{\x1a\xde\xee\xe8\xcc\xe6s\xceS\xa8\xa0\xc8~\x03B\xe6\x19i\xe1\xeb\x87\xbb\xdc=\x00\x80\x03}q\x00(X\x08\x00\x00\x00is_retryq\x01\x89X\t\x00\x00\x00operationq\x02}q\x03(X\x02\x00\x00\x00idq\x04K\tX\t\x00\x00\x00client_idq\x05X\x02\x00\x00\x00c0q\x06X\x0e\x00\x00\x00command_stringq\x07X\x0c\x00\x00\x00get('movie')q\x08uu."}
[2017-11-21 01:49:23,813] replica.Replica<Replica:98002>:OUTPUT: Received result shuttle (source, client_id, operation_id, self_id)  r1 c0 8 r0
[2017-11-21 01:49:23,813] replica.Replica<Replica:98002>:OUTPUT: Checking for failure associated with trigger, application:  ('result_shuttle', 'pre')  in replica:  r0
[2017-11-21 01:49:23,814] replica.Replica<Replica:98002>:OUTPUT: No failure associated with trigger:  result_shuttle  found in replica:  r0
[2017-11-21 01:49:23,826] replica.Replica<Replica:98002>:OUTPUT: Received request message (source, client_id, operation_id, self_id)  c0 c0 9 r0
[2017-11-21 01:49:23,827] replica.Replica<Replica:98002>:OUTPUT: Checking for failure associated with trigger, application:  ('client_request', 'pre')  in replica:  r0
[2017-11-21 01:49:23,827] replica.Replica<Replica:98002>:OUTPUT: No failure associated with trigger:  client_request  found in replica:  r0
[2017-11-21 01:49:23,831] replica.Replica<Replica:98002>:OUTPUT: Head is processing request (operation_id, client_id) (9, 'c0')
[2017-11-21 01:49:23,837] replica.Replica<Replica:98002>:OUTPUT: Checking for failure associated with trigger, application:  ('client_request', 'post')  in replica:  r0
[2017-11-21 01:49:23,838] replica.Replica<Replica:98002>:OUTPUT: No failure associated with trigger:  client_request  found in replica:  r0
[2017-11-21 01:49:23,840] replica.Replica<Replica:98003>:OUTPUT: Received shuttle (source, client_id, operation_id, self_id)  r0 c0 9 r1
[2017-11-21 01:49:23,841] replica.Replica<Replica:98003>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'pre')  in replica:  r1
[2017-11-21 01:49:23,841] replica.Replica<Replica:98003>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r1
[2017-11-21 01:49:23,857] replica.Replica<Replica:98003>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'post')  in replica:  r1
[2017-11-21 01:49:23,857] replica.Replica<Replica:98003>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r1
[2017-11-21 01:49:23,861] replica.Replica<Replica:98004>:OUTPUT: Received shuttle (source, client_id, operation_id, self_id)  r1 c0 9 r2
[2017-11-21 01:49:23,861] replica.Replica<Replica:98004>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'pre')  in replica:  r2
[2017-11-21 01:49:23,862] replica.Replica<Replica:98004>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r2
[2017-11-21 01:49:23,885] replica.Replica<Replica:98004>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'post')  in replica:  r2
[2017-11-21 01:49:23,885] replica.Replica<Replica:98004>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r2
[2017-11-21 01:49:23,888] replica.Replica<Replica:98003>:OUTPUT: Received result shuttle (source, client_id, operation_id, self_id)  r2 c0 9 r1
[2017-11-21 01:49:23,888] replica.Replica<Replica:98004>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'post')  in replica:  r2
[2017-11-21 01:49:23,889] replica.Replica<Replica:98003>:OUTPUT: Checking for failure associated with trigger, application:  ('result_shuttle', 'pre')  in replica:  r1
[2017-11-21 01:49:23,889] replica.Replica<Replica:98004>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r2
[2017-11-21 01:49:23,889] replica.Replica<Replica:98003>:OUTPUT: No failure associated with trigger:  result_shuttle  found in replica:  r1
[2017-11-21 01:49:23,891] client.Client<Client:98001>:OUTPUT: A result was received from replica= r2
[2017-11-21 01:49:23,906] client.Client<Client:98001>:OUTPUT: The operation result pair is= {'id': 9, 'client_id': 'c0', 'command_string': "get('movie')"} star123
[2017-11-21 01:49:23,907] client.Client<Client:98001>:OUTPUT: The result received is valid:(result, operation)= star123 {'id': 9, 'client_id': 'c0', 'command_string': "get('movie')"}
[2017-11-21 01:49:23,907] replica.Replica<Replica:98003>:OUTPUT: Checking for failure associated with trigger, application:  ('result_shuttle', 'post')  in replica:  r1
[2017-11-21 01:49:23,907] client.Client<Client:98001>:OUTPUT: The following operation was created: {'id': 10, 'client_id': 'c0', 'command_string': "get('movie')"}
[2017-11-21 01:49:23,907] replica.Replica<Replica:98003>:OUTPUT: No failure associated with trigger:  result_shuttle  found in replica:  r1
[2017-11-21 01:49:23,908] client.Client<Client:98001>:OUTPUT: return_value= star123
[2017-11-21 01:49:23,911] client.Client<Client:98001>:OUTPUT: A request for an operation was made {'content': {'is_retry': False, 'operation': {'id': 10, 'client_id': 'c0', 'command_string': "get('movie')"}}, 'signed_content': b"\xce\xcf\x18\x00H'\x99\xd8\x18\xf7\xaf\xb5\x0f\xb1\xab\x7f\xac\xf8d^\x10\x14\xcb\x90\x16b\xdaut\x84\xc0aS\xbe\xe8B`\x94\x11\x07K+I\xf3\xfe]\xee\xc6\x1eh?\xc6\xde\xaa\x9c\xe2~\x0f\xd6\xe0l\xb7\x06\x08\x80\x03}q\x00(X\x08\x00\x00\x00is_retryq\x01\x89X\t\x00\x00\x00operationq\x02}q\x03(X\x02\x00\x00\x00idq\x04K\nX\t\x00\x00\x00client_idq\x05X\x02\x00\x00\x00c0q\x06X\x0e\x00\x00\x00command_stringq\x07X\x0c\x00\x00\x00get('movie')q\x08uu."}
[2017-11-21 01:49:23,912] replica.Replica<Replica:98002>:OUTPUT: Received result shuttle (source, client_id, operation_id, self_id)  r1 c0 9 r0
[2017-11-21 01:49:23,913] replica.Replica<Replica:98002>:OUTPUT: Checking for failure associated with trigger, application:  ('result_shuttle', 'pre')  in replica:  r0
[2017-11-21 01:49:23,914] replica.Replica<Replica:98002>:OUTPUT: No failure associated with trigger:  result_shuttle  found in replica:  r0
[2017-11-21 01:49:23,926] replica.Replica<Replica:98002>:OUTPUT: Received request message (source, client_id, operation_id, self_id)  c0 c0 10 r0
[2017-11-21 01:49:23,927] replica.Replica<Replica:98002>:OUTPUT: Checking for failure associated with trigger, application:  ('client_request', 'pre')  in replica:  r0
[2017-11-21 01:49:23,927] replica.Replica<Replica:98002>:OUTPUT: No failure associated with trigger:  client_request  found in replica:  r0
[2017-11-21 01:49:23,930] replica.Replica<Replica:98002>:OUTPUT: Head is processing request (operation_id, client_id) (10, 'c0')
[2017-11-21 01:49:23,935] replica.Replica<Replica:98002>:OUTPUT: Checking for failure associated with trigger, application:  ('client_request', 'post')  in replica:  r0
[2017-11-21 01:49:23,936] replica.Replica<Replica:98002>:OUTPUT: No failure associated with trigger:  client_request  found in replica:  r0
[2017-11-21 01:49:23,939] replica.Replica<Replica:98003>:OUTPUT: Received shuttle (source, client_id, operation_id, self_id)  r0 c0 10 r1
[2017-11-21 01:49:23,939] replica.Replica<Replica:98003>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'pre')  in replica:  r1
[2017-11-21 01:49:23,939] replica.Replica<Replica:98003>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r1
[2017-11-21 01:49:23,953] replica.Replica<Replica:98003>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'post')  in replica:  r1
[2017-11-21 01:49:23,953] replica.Replica<Replica:98003>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r1
[2017-11-21 01:49:23,957] replica.Replica<Replica:98004>:OUTPUT: Received shuttle (source, client_id, operation_id, self_id)  r1 c0 10 r2
[2017-11-21 01:49:23,957] replica.Replica<Replica:98004>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'pre')  in replica:  r2
[2017-11-21 01:49:23,958] replica.Replica<Replica:98004>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r2
[2017-11-21 01:49:23,984] replica.Replica<Replica:98004>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'post')  in replica:  r2
[2017-11-21 01:49:23,984] replica.Replica<Replica:98004>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r2
[2017-11-21 01:49:23,987] replica.Replica<Replica:98003>:OUTPUT: Received result shuttle (source, client_id, operation_id, self_id)  r2 c0 10 r1
[2017-11-21 01:49:23,987] replica.Replica<Replica:98004>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'post')  in replica:  r2
[2017-11-21 01:49:23,987] replica.Replica<Replica:98003>:OUTPUT: Checking for failure associated with trigger, application:  ('result_shuttle', 'pre')  in replica:  r1
[2017-11-21 01:49:23,987] replica.Replica<Replica:98004>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r2
[2017-11-21 01:49:23,988] replica.Replica<Replica:98003>:OUTPUT: No failure associated with trigger:  result_shuttle  found in replica:  r1
[2017-11-21 01:49:23,989] client.Client<Client:98001>:OUTPUT: A result was received from replica= r2
[2017-11-21 01:49:24,000] client.Client<Client:98001>:OUTPUT: The operation result pair is= {'id': 10, 'client_id': 'c0', 'command_string': "get('movie')"} star123
[2017-11-21 01:49:24,001] client.Client<Client:98001>:OUTPUT: The result received is valid:(result, operation)= star123 {'id': 10, 'client_id': 'c0', 'command_string': "get('movie')"}
[2017-11-21 01:49:24,001] client.Client<Client:98001>:OUTPUT: The following operation was created: {'id': 11, 'client_id': 'c0', 'command_string': "get('movie')"}
[2017-11-21 01:49:24,002] client.Client<Client:98001>:OUTPUT: return_value= star123
[2017-11-21 01:49:24,003] replica.Replica<Replica:98003>:OUTPUT: Checking for failure associated with trigger, application:  ('result_shuttle', 'post')  in replica:  r1
[2017-11-21 01:49:24,003] replica.Replica<Replica:98003>:OUTPUT: No failure associated with trigger:  result_shuttle  found in replica:  r1
[2017-11-21 01:49:24,005] client.Client<Client:98001>:OUTPUT: A request for an operation was made {'content': {'is_retry': False, 'operation': {'id': 11, 'client_id': 'c0', 'command_string': "get('movie')"}}, 'signed_content': b'.-"M\xbf\xf5>\xee\x83\xcb>\xb9\xdb\x87>R\xdbY\xfdSK\xfa1?Uge*\x9ai\xda\x13\xcbg\x1b\x87.\xad\x07\x82\xc8\xa3\xf4\xbd\x18C\xd8FX\xba\x1f\x04F\xbf\nT\xe0\xac\x97"\xb3m\xe4\x0c\x80\x03}q\x00(X\x08\x00\x00\x00is_retryq\x01\x89X\t\x00\x00\x00operationq\x02}q\x03(X\x02\x00\x00\x00idq\x04K\x0bX\t\x00\x00\x00client_idq\x05X\x02\x00\x00\x00c0q\x06X\x0e\x00\x00\x00command_stringq\x07X\x0c\x00\x00\x00get(\'movie\')q\x08uu.'}
[2017-11-21 01:49:24,007] replica.Replica<Replica:98002>:OUTPUT: Received request message (source, client_id, operation_id, self_id)  c0 c0 11 r0
[2017-11-21 01:49:24,007] replica.Replica<Replica:98002>:OUTPUT: Checking for failure associated with trigger, application:  ('client_request', 'pre')  in replica:  r0
[2017-11-21 01:49:24,008] replica.Replica<Replica:98002>:OUTPUT: No failure associated with trigger:  client_request  found in replica:  r0
[2017-11-21 01:49:24,012] replica.Replica<Replica:98002>:OUTPUT: Head is processing request (operation_id, client_id) (11, 'c0')
[2017-11-21 01:49:24,017] replica.Replica<Replica:98002>:OUTPUT: Checking for failure associated with trigger, application:  ('client_request', 'post')  in replica:  r0
[2017-11-21 01:49:24,017] replica.Replica<Replica:98002>:OUTPUT: No failure associated with trigger:  client_request  found in replica:  r0
[2017-11-21 01:49:24,019] replica.Replica<Replica:98002>:OUTPUT: Received result shuttle (source, client_id, operation_id, self_id)  r1 c0 10 r0
[2017-11-21 01:49:24,019] replica.Replica<Replica:98003>:OUTPUT: Received shuttle (source, client_id, operation_id, self_id)  r0 c0 11 r1
[2017-11-21 01:49:24,020] replica.Replica<Replica:98002>:OUTPUT: Checking for failure associated with trigger, application:  ('result_shuttle', 'pre')  in replica:  r0
[2017-11-21 01:49:24,020] replica.Replica<Replica:98003>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'pre')  in replica:  r1
[2017-11-21 01:49:24,020] replica.Replica<Replica:98002>:OUTPUT: No failure associated with trigger:  result_shuttle  found in replica:  r0
[2017-11-21 01:49:24,021] replica.Replica<Replica:98003>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r1
[2017-11-21 01:49:24,039] replica.Replica<Replica:98003>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'post')  in replica:  r1
[2017-11-21 01:49:24,039] replica.Replica<Replica:98003>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r1
[2017-11-21 01:49:24,043] replica.Replica<Replica:98004>:OUTPUT: Received shuttle (source, client_id, operation_id, self_id)  r1 c0 11 r2
[2017-11-21 01:49:24,043] replica.Replica<Replica:98004>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'pre')  in replica:  r2
[2017-11-21 01:49:24,043] replica.Replica<Replica:98004>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r2
[2017-11-21 01:49:24,065] replica.Replica<Replica:98004>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'post')  in replica:  r2
[2017-11-21 01:49:24,065] replica.Replica<Replica:98004>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r2
[2017-11-21 01:49:24,068] replica.Replica<Replica:98003>:OUTPUT: Received result shuttle (source, client_id, operation_id, self_id)  r2 c0 11 r1
[2017-11-21 01:49:24,068] replica.Replica<Replica:98004>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'post')  in replica:  r2
[2017-11-21 01:49:24,069] replica.Replica<Replica:98003>:OUTPUT: Checking for failure associated with trigger, application:  ('result_shuttle', 'pre')  in replica:  r1
[2017-11-21 01:49:24,069] replica.Replica<Replica:98004>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r2
[2017-11-21 01:49:24,069] replica.Replica<Replica:98003>:OUTPUT: No failure associated with trigger:  result_shuttle  found in replica:  r1
[2017-11-21 01:49:24,071] client.Client<Client:98001>:OUTPUT: A result was received from replica= r2
[2017-11-21 01:49:24,084] client.Client<Client:98001>:OUTPUT: The operation result pair is= {'id': 11, 'client_id': 'c0', 'command_string': "get('movie')"} star123
[2017-11-21 01:49:24,085] client.Client<Client:98001>:OUTPUT: The result received is valid:(result, operation)= star123 {'id': 11, 'client_id': 'c0', 'command_string': "get('movie')"}
[2017-11-21 01:49:24,086] client.Client<Client:98001>:OUTPUT: The following operation was created: {'id': 12, 'client_id': 'c0', 'command_string': "get('movie')"}
[2017-11-21 01:49:24,086] replica.Replica<Replica:98003>:OUTPUT: Checking for failure associated with trigger, application:  ('result_shuttle', 'post')  in replica:  r1
[2017-11-21 01:49:24,086] client.Client<Client:98001>:OUTPUT: return_value= star123
[2017-11-21 01:49:24,087] replica.Replica<Replica:98003>:OUTPUT: No failure associated with trigger:  result_shuttle  found in replica:  r1
[2017-11-21 01:49:24,089] client.Client<Client:98001>:OUTPUT: A request for an operation was made {'content': {'is_retry': False, 'operation': {'id': 12, 'client_id': 'c0', 'command_string': "get('movie')"}}, 'signed_content': b'\x9b\x84\t\x83\xbe--"\x18\xd7E\x00\xf7?\xa5\xaetX\x97\xb4\x11\x8b\xfctK\xc5\x83\n\r\xe3\x9b\x179\xd2\x8a\xf4PP\xa5\x11\xb5A\x1d\x1eL_\xb9]\x9e\r\xb51\xcd\xfb\x85\xfbfg\xc7\x8f)x\xdf\x02\x80\x03}q\x00(X\x08\x00\x00\x00is_retryq\x01\x89X\t\x00\x00\x00operationq\x02}q\x03(X\x02\x00\x00\x00idq\x04K\x0cX\t\x00\x00\x00client_idq\x05X\x02\x00\x00\x00c0q\x06X\x0e\x00\x00\x00command_stringq\x07X\x0c\x00\x00\x00get(\'movie\')q\x08uu.'}
[2017-11-21 01:49:24,091] replica.Replica<Replica:98002>:OUTPUT: Received result shuttle (source, client_id, operation_id, self_id)  r1 c0 11 r0
[2017-11-21 01:49:24,091] replica.Replica<Replica:98002>:OUTPUT: Checking for failure associated with trigger, application:  ('result_shuttle', 'pre')  in replica:  r0
[2017-11-21 01:49:24,092] replica.Replica<Replica:98002>:OUTPUT: No failure associated with trigger:  result_shuttle  found in replica:  r0
[2017-11-21 01:49:24,107] replica.Replica<Replica:98002>:OUTPUT: Received request message (source, client_id, operation_id, self_id)  c0 c0 12 r0
[2017-11-21 01:49:24,108] replica.Replica<Replica:98002>:OUTPUT: Checking for failure associated with trigger, application:  ('client_request', 'pre')  in replica:  r0
[2017-11-21 01:49:24,108] replica.Replica<Replica:98002>:OUTPUT: No failure associated with trigger:  client_request  found in replica:  r0
[2017-11-21 01:49:24,111] replica.Replica<Replica:98002>:OUTPUT: Head is processing request (operation_id, client_id) (12, 'c0')
[2017-11-21 01:49:24,116] replica.Replica<Replica:98002>:OUTPUT: Checking for failure associated with trigger, application:  ('client_request', 'post')  in replica:  r0
[2017-11-21 01:49:24,117] replica.Replica<Replica:98002>:OUTPUT: No failure associated with trigger:  client_request  found in replica:  r0
[2017-11-21 01:49:24,120] replica.Replica<Replica:98003>:OUTPUT: Received shuttle (source, client_id, operation_id, self_id)  r0 c0 12 r1
[2017-11-21 01:49:24,120] replica.Replica<Replica:98003>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'pre')  in replica:  r1
[2017-11-21 01:49:24,121] replica.Replica<Replica:98003>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r1
[2017-11-21 01:49:24,133] replica.Replica<Replica:98003>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'post')  in replica:  r1
[2017-11-21 01:49:24,134] replica.Replica<Replica:98003>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r1
[2017-11-21 01:49:24,136] replica.Replica<Replica:98004>:OUTPUT: Received shuttle (source, client_id, operation_id, self_id)  r1 c0 12 r2
[2017-11-21 01:49:24,137] replica.Replica<Replica:98004>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'pre')  in replica:  r2
[2017-11-21 01:49:24,137] replica.Replica<Replica:98004>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r2
[2017-11-21 01:49:24,161] replica.Replica<Replica:98004>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'post')  in replica:  r2
[2017-11-21 01:49:24,161] replica.Replica<Replica:98004>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r2
[2017-11-21 01:49:24,165] replica.Replica<Replica:98004>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'post')  in replica:  r2
[2017-11-21 01:49:24,165] replica.Replica<Replica:98003>:OUTPUT: Received result shuttle (source, client_id, operation_id, self_id)  r2 c0 12 r1
[2017-11-21 01:49:24,165] replica.Replica<Replica:98004>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r2
[2017-11-21 01:49:24,165] replica.Replica<Replica:98003>:OUTPUT: Checking for failure associated with trigger, application:  ('result_shuttle', 'pre')  in replica:  r1
[2017-11-21 01:49:24,166] replica.Replica<Replica:98003>:OUTPUT: No failure associated with trigger:  result_shuttle  found in replica:  r1
[2017-11-21 01:49:24,167] client.Client<Client:98001>:OUTPUT: A result was received from replica= r2
[2017-11-21 01:49:24,182] replica.Replica<Replica:98003>:OUTPUT: Checking for failure associated with trigger, application:  ('result_shuttle', 'post')  in replica:  r1
[2017-11-21 01:49:24,182] replica.Replica<Replica:98003>:OUTPUT: No failure associated with trigger:  result_shuttle  found in replica:  r1
[2017-11-21 01:49:24,183] client.Client<Client:98001>:OUTPUT: The operation result pair is= {'id': 12, 'client_id': 'c0', 'command_string': "get('movie')"} star123
[2017-11-21 01:49:24,183] client.Client<Client:98001>:OUTPUT: The result received is valid:(result, operation)= star123 {'id': 12, 'client_id': 'c0', 'command_string': "get('movie')"}
[2017-11-21 01:49:24,184] client.Client<Client:98001>:OUTPUT: The following operation was created: {'id': 13, 'client_id': 'c0', 'command_string': "get('movie')"}
[2017-11-21 01:49:24,184] client.Client<Client:98001>:OUTPUT: return_value= star123
[2017-11-21 01:49:24,187] replica.Replica<Replica:98002>:OUTPUT: Received result shuttle (source, client_id, operation_id, self_id)  r1 c0 12 r0
[2017-11-21 01:49:24,188] client.Client<Client:98001>:OUTPUT: A request for an operation was made {'content': {'is_retry': False, 'operation': {'id': 13, 'client_id': 'c0', 'command_string': "get('movie')"}}, 'signed_content': b'rA\xe5\xc7\x8a\xe2\x15\x1do\tB\x7fK B\x83\x86}\xe9\xb6\xd8\xdb\xdfS\x14,\xa7h\xed\xdbA\xd3\xc4I\nu\xf7\x82\x1c6O\x05\xc8\x97\x83\xbe\x18\xafH\x02\xe7R"\xb2\x7fz\xa4\xf1\xe4{\xda\xe3%\x00\x80\x03}q\x00(X\x08\x00\x00\x00is_retryq\x01\x89X\t\x00\x00\x00operationq\x02}q\x03(X\x02\x00\x00\x00idq\x04K\rX\t\x00\x00\x00client_idq\x05X\x02\x00\x00\x00c0q\x06X\x0e\x00\x00\x00command_stringq\x07X\x0c\x00\x00\x00get(\'movie\')q\x08uu.'}
[2017-11-21 01:49:24,188] replica.Replica<Replica:98002>:OUTPUT: Checking for failure associated with trigger, application:  ('result_shuttle', 'pre')  in replica:  r0
[2017-11-21 01:49:24,189] replica.Replica<Replica:98002>:OUTPUT: No failure associated with trigger:  result_shuttle  found in replica:  r0
[2017-11-21 01:49:24,203] replica.Replica<Replica:98002>:OUTPUT: Received request message (source, client_id, operation_id, self_id)  c0 c0 13 r0
[2017-11-21 01:49:24,203] replica.Replica<Replica:98002>:OUTPUT: Checking for failure associated with trigger, application:  ('client_request', 'pre')  in replica:  r0
[2017-11-21 01:49:24,203] replica.Replica<Replica:98002>:OUTPUT: No failure associated with trigger:  client_request  found in replica:  r0
[2017-11-21 01:49:24,206] replica.Replica<Replica:98002>:OUTPUT: Head is processing request (operation_id, client_id) (13, 'c0')
[2017-11-21 01:49:24,213] replica.Replica<Replica:98002>:OUTPUT: Checking for failure associated with trigger, application:  ('client_request', 'post')  in replica:  r0
[2017-11-21 01:49:24,214] replica.Replica<Replica:98002>:OUTPUT: No failure associated with trigger:  client_request  found in replica:  r0
[2017-11-21 01:49:24,217] replica.Replica<Replica:98003>:OUTPUT: Received shuttle (source, client_id, operation_id, self_id)  r0 c0 13 r1
[2017-11-21 01:49:24,217] replica.Replica<Replica:98003>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'pre')  in replica:  r1
[2017-11-21 01:49:24,217] replica.Replica<Replica:98003>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r1
[2017-11-21 01:49:24,233] replica.Replica<Replica:98003>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'post')  in replica:  r1
[2017-11-21 01:49:24,234] replica.Replica<Replica:98003>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r1
[2017-11-21 01:49:24,237] replica.Replica<Replica:98004>:OUTPUT: Received shuttle (source, client_id, operation_id, self_id)  r1 c0 13 r2
[2017-11-21 01:49:24,237] replica.Replica<Replica:98004>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'pre')  in replica:  r2
[2017-11-21 01:49:24,237] replica.Replica<Replica:98004>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r2
[2017-11-21 01:49:24,264] replica.Replica<Replica:98004>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'post')  in replica:  r2
[2017-11-21 01:49:24,264] replica.Replica<Replica:98004>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r2
[2017-11-21 01:49:24,267] replica.Replica<Replica:98003>:OUTPUT: Received result shuttle (source, client_id, operation_id, self_id)  r2 c0 13 r1
[2017-11-21 01:49:24,267] replica.Replica<Replica:98004>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'post')  in replica:  r2
[2017-11-21 01:49:24,267] replica.Replica<Replica:98003>:OUTPUT: Checking for failure associated with trigger, application:  ('result_shuttle', 'pre')  in replica:  r1
[2017-11-21 01:49:24,268] replica.Replica<Replica:98004>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r2
[2017-11-21 01:49:24,268] replica.Replica<Replica:98003>:OUTPUT: No failure associated with trigger:  result_shuttle  found in replica:  r1
[2017-11-21 01:49:24,270] client.Client<Client:98001>:OUTPUT: A result was received from replica= r2
[2017-11-21 01:49:24,285] client.Client<Client:98001>:OUTPUT: The operation result pair is= {'id': 13, 'client_id': 'c0', 'command_string': "get('movie')"} star123
[2017-11-21 01:49:24,285] replica.Replica<Replica:98003>:OUTPUT: Checking for failure associated with trigger, application:  ('result_shuttle', 'post')  in replica:  r1
[2017-11-21 01:49:24,286] client.Client<Client:98001>:OUTPUT: The result received is valid:(result, operation)= star123 {'id': 13, 'client_id': 'c0', 'command_string': "get('movie')"}
[2017-11-21 01:49:24,286] replica.Replica<Replica:98003>:OUTPUT: No failure associated with trigger:  result_shuttle  found in replica:  r1
[2017-11-21 01:49:24,287] client.Client<Client:98001>:OUTPUT: The following operation was created: {'id': 14, 'client_id': 'c0', 'command_string': "get('movie');"}
[2017-11-21 01:49:24,287] client.Client<Client:98001>:OUTPUT: return_value= star123
[2017-11-21 01:49:24,291] replica.Replica<Replica:98002>:OUTPUT: Received result shuttle (source, client_id, operation_id, self_id)  r1 c0 13 r0
[2017-11-21 01:49:24,291] client.Client<Client:98001>:OUTPUT: A request for an operation was made {'content': {'is_retry': False, 'operation': {'id': 14, 'client_id': 'c0', 'command_string': "get('movie');"}}, 'signed_content': b"\xdb\x1c\xc4P\xb4@|P\xe6\xcf\xb0[%\xc7&\x18i\xabi\n5d,g\xe9f\xbe*\x00\xb4\xc3\xe4a\xdb\x1a\xeb\xdd\xad\xb1\x89\x11KW\xa7\xd5\xb3\xd2\xba\xbe\xf5$\x1b\xa3\xbd\xb3\xca\xe9F\xcd\xbb\x83\xd8k\t\x80\x03}q\x00(X\x08\x00\x00\x00is_retryq\x01\x89X\t\x00\x00\x00operationq\x02}q\x03(X\x02\x00\x00\x00idq\x04K\x0eX\t\x00\x00\x00client_idq\x05X\x02\x00\x00\x00c0q\x06X\x0e\x00\x00\x00command_stringq\x07X\r\x00\x00\x00get('movie');q\x08uu."}
[2017-11-21 01:49:24,292] replica.Replica<Replica:98002>:OUTPUT: Checking for failure associated with trigger, application:  ('result_shuttle', 'pre')  in replica:  r0
[2017-11-21 01:49:24,293] replica.Replica<Replica:98002>:OUTPUT: No failure associated with trigger:  result_shuttle  found in replica:  r0
[2017-11-21 01:49:24,308] replica.Replica<Replica:98002>:OUTPUT: Received request message (source, client_id, operation_id, self_id)  c0 c0 14 r0
[2017-11-21 01:49:24,308] replica.Replica<Replica:98002>:OUTPUT: Checking for failure associated with trigger, application:  ('client_request', 'pre')  in replica:  r0
[2017-11-21 01:49:24,309] replica.Replica<Replica:98002>:OUTPUT: No failure associated with trigger:  client_request  found in replica:  r0
[2017-11-21 01:49:24,313] replica.Replica<Replica:98002>:OUTPUT: Head is processing request (operation_id, client_id) (14, 'c0')
[2017-11-21 01:49:24,318] replica.Replica<Replica:98002>:OUTPUT: Checking for failure associated with trigger, application:  ('client_request', 'post')  in replica:  r0
[2017-11-21 01:49:24,319] replica.Replica<Replica:98002>:OUTPUT: No failure associated with trigger:  client_request  found in replica:  r0
[2017-11-21 01:49:24,321] replica.Replica<Replica:98003>:OUTPUT: Received shuttle (source, client_id, operation_id, self_id)  r0 c0 14 r1
[2017-11-21 01:49:24,321] replica.Replica<Replica:98003>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'pre')  in replica:  r1
[2017-11-21 01:49:24,322] replica.Replica<Replica:98003>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r1
[2017-11-21 01:49:24,338] replica.Replica<Replica:98003>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'post')  in replica:  r1
[2017-11-21 01:49:24,339] replica.Replica<Replica:98003>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r1
[2017-11-21 01:49:24,343] replica.Replica<Replica:98004>:OUTPUT: Received shuttle (source, client_id, operation_id, self_id)  r1 c0 14 r2
[2017-11-21 01:49:24,343] replica.Replica<Replica:98004>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'pre')  in replica:  r2
[2017-11-21 01:49:24,344] replica.Replica<Replica:98004>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r2
[2017-11-21 01:49:24,376] replica.Replica<Replica:98004>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'post')  in replica:  r2
[2017-11-21 01:49:24,376] replica.Replica<Replica:98004>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r2
[2017-11-21 01:49:24,380] replica.Replica<Replica:98003>:OUTPUT: Received result shuttle (source, client_id, operation_id, self_id)  r2 c0 14 r1
[2017-11-21 01:49:24,381] replica.Replica<Replica:98003>:OUTPUT: Checking for failure associated with trigger, application:  ('result_shuttle', 'pre')  in replica:  r1
[2017-11-21 01:49:24,381] replica.Replica<Replica:98003>:OUTPUT: No failure associated with trigger:  result_shuttle  found in replica:  r1
[2017-11-21 01:49:24,382] replica.Replica<Replica:98004>:OUTPUT: Checking for failure associated with trigger, application:  ('shuttle', 'post')  in replica:  r2
[2017-11-21 01:49:24,383] replica.Replica<Replica:98004>:OUTPUT: No failure associated with trigger:  shuttle  found in replica:  r2
[2017-11-21 01:49:24,386] client.Client<Client:98001>:OUTPUT: A result was received from replica= r2
[2017-11-21 01:49:24,397] client.Client<Client:98001>:OUTPUT: The operation result pair is= {'id': 14, 'client_id': 'c0', 'command_string': "get('movie');"} star123
[2017-11-21 01:49:24,398] client.Client<Client:98001>:OUTPUT: The result received is valid:(result, operation)= star123 {'id': 14, 'client_id': 'c0', 'command_string': "get('movie');"}
[2017-11-21 01:49:24,398] client.Client<Client:98001>:OUTPUT: ~~~~~~~~~~Final state: {'movie': 'star123'}
[2017-11-21 01:49:24,398] client.Client<Client:98001>:OUTPUT: Requesting shared object from head from client: c0
[2017-11-21 01:49:24,400] replica.Replica<Replica:98003>:OUTPUT: Checking for failure associated with trigger, application:  ('result_shuttle', 'post')  in replica:  r1
[2017-11-21 01:49:24,400] replica.Replica<Replica:98003>:OUTPUT: No failure associated with trigger:  result_shuttle  found in replica:  r1
[2017-11-21 01:49:24,402] replica.Replica<Replica:98002>:OUTPUT: Received wedge request from Olympus in Replica:  r0
[2017-11-21 01:49:24,405] replica.Replica<Replica:98002>:OUTPUT: Received shared_object_request from client= c0
[2017-11-21 01:49:24,411] client.Client<Client:98001>:OUTPUT: Received shared object from head =  {'movie': 'star123'}
[2017-11-21 01:49:24,412] replica.Replica<Replica:98002>:OUTPUT: Received result shuttle (source, client_id, operation_id, self_id)  r1 c0 14 r0
[2017-11-21 01:49:24,413] replica.Replica<Replica:98002>:OUTPUT: Checking for failure associated with trigger, application:  ('result_shuttle', 'pre')  in replica:  r0
[2017-11-21 01:49:24,413] replica.Replica<Replica:98002>:OUTPUT: No failure associated with trigger:  result_shuttle  found in replica:  r0
[2017-11-21 01:49:24,415] client.Client<Client:98001>:OUTPUT: Final results are consistent
[2017-11-21 01:49:24,416] client.Client<Client:98001>:OUTPUT: Workload finished for client:  c0
[2017-11-21 01:49:24,416] client.Client<Client:98001>:OUTPUT: Sending done message to Olympus and terminating
[2017-11-21 01:49:24,421] olympus.Olympus<Olympus:e4002>:OUTPUT: Done message received from client:  c0
[2017-11-21 01:49:24,426] olympus.Olympus<Olympus:e4002>:OUTPUT: Terminating replicas..
[2017-11-21 01:49:24,431] olympus.Olympus<Olympus:e4002>:OUTPUT: Sent abort command to replica : r0
[2017-11-21 01:49:24,433] replica.Replica<Replica:98002>:OUTPUT: Received abort message from Olympus in replica:  r0
[2017-11-21 01:49:24,436] olympus.Olympus<Olympus:e4002>:OUTPUT: Sent abort command to replica : r1
[2017-11-21 01:49:24,438] replica.Replica<Replica:98002>:OUTPUT: Sending aborted message to Olympus from:  r0
[2017-11-21 01:49:24,438] replica.Replica<Replica:98003>:OUTPUT: Received abort message from Olympus in replica:  r1
[2017-11-21 01:49:24,440] olympus.Olympus<Olympus:e4002>:OUTPUT: Sent abort command to replica : r2
[2017-11-21 01:49:24,440] olympus.Olympus<Olympus:e4002>:OUTPUT: Olympus terminating
[2017-11-21 01:49:24,441] replica.Replica<Replica:98002>:OUTPUT: Shutting down replica:  r0
[2017-11-21 01:49:24,441] replica.Replica<Replica:98004>:OUTPUT: Received abort message from Olympus in replica:  r2
[2017-11-21 01:49:24,442] olympus.Olympus<Olympus:e4002>:OUTPUT: elapsed time (seconds):  2.53664231300354
[2017-11-21 01:49:24,442] replica.Replica<Replica:98003>:OUTPUT: Sending aborted message to Olympus from:  r1
[2017-11-21 01:49:24,445] replica.Replica<Replica:98003>:OUTPUT: Shutting down replica:  r1
[2017-11-21 01:49:24,445] replica.Replica<Replica:98004>:OUTPUT: Sending aborted message to Olympus from:  r2
[2017-11-21 01:49:24,451] replica.Replica<Replica:98004>:OUTPUT: Shutting down replica:  r2
